src/main/java/com/github/rnewson/couchdb/lucene/util/Analyzers.java

```diff
@@ -186,13 +280,313 @@ protected TokenStreamComponents wrapComponents(String fieldName, TokenStreamComp
         final String[] parts = str.split(":", 2);
         final String name = parts[0].toUpperCase();
         final String args = parts.length == 2 ? parts[1] : null;
         return Analyzers.valueOf(name).newAnalyzer(args);
     }
 
+	public static Analyzer getAnalyzer(final JSONObject json) {
+		String className = json.optString(Constants.CLASS);
+		JSONArray params = json.optJSONArray(Constants.PARAMS);
+
+		Analyzer newAnalyzer = null;
+
+		if (className == null || className.isEmpty()) {
+			Iterator<?> it = json.keys();
+			if (it.hasNext()) {
+				String key = (String) it.next();
+				String args = json.optString(key);
+				
+				JSONObject obj = json.optJSONObject(key);
+				try {
+					if (obj != null) {
+						return Analyzers.valueOf(key.toUpperCase()).newAnalyzer(obj);
+					} else {
+						return Analyzers.valueOf(key.toUpperCase()).newAnalyzer(args);
+					}
+				} catch (Exception ex) { }
+			}
+
+			logger.error("Lucene index: analyzer class name is not defined");
+			return null;
+		}
+
+		// is the class accessible?
+		Class<?> clazz = null;
+		try {
+			clazz = Class.forName(className);
+		} catch (ClassNotFoundException e) {
+			logger.error(String.format("Lucene index: analyzer class %s not found. (%s)", className, e.getMessage()));
+			return null;
+		}
+
+		// Is the class an Analyzer?
+		if (!Analyzer.class.isAssignableFrom(clazz)) {
+			logger.error(String.format("Lucene index: analyzer class has to be a subclass of %s", Analyzer.class.getName()));
+			return null;
+		}
+
+		// Get list of parameters
+		List<KeyTypedValue> cParams;
+		try {
+			cParams = getAllConstructorParameters(params);
+		} catch (Exception pe) {
+			logger.error(String.format("Unable to get parameters for %s: %s", className, pe.getMessage()), pe);
+			cParams = new ArrayList<>();
+		}
+
+		// Iterate over all parameters, convert data to two arrays
+		// that can be used in the reflection code
+		final Class<?> cParamClasses[] = new Class<?>[cParams.size()];
+		final Object cParamValues[] = new Object[cParams.size()];
+		for (int i = 0; i < cParams.size(); i++) {
+			KeyTypedValue ktv = cParams.get(i);
+			cParamClasses[i] = ktv.getValueClass();
+			cParamValues[i] = ktv.getValue();
+		}
+
+		// Create new analyzer
+		newAnalyzer = createInstance(clazz, cParamClasses, cParamValues);
+
+		if (newAnalyzer == null) {
+			logger.error(String.format("Unable to create analyzer '%s'", className));
+		}
+		
+		return newAnalyzer;
+	}
+
+	/**
+	 * Create instance of the lucene analyzer with provided arguments
+	 *
+	 * @param clazz The analyzer class
+	 * @param vcParamClasses The parameter classes
+	 * @param vcParamValues The parameter values
+	 * @return The lucene analyzer
+	 */
+	private static Analyzer createInstance(Class<?> clazz, Class<?>[] vcParamClasses, Object[] vcParamValues) {
+
+		String className = clazz.getName();
+
+		try {
+			final Constructor<?> cstr = clazz.getDeclaredConstructor(vcParamClasses);
+			cstr.setAccessible(true);
```

Comment:
it's been a while since I delved into reflection, do we need to setAccessible(true) here?